packages/crayons-core/src/components/data-table/data-table.tsx

```diff
@@ -333,8 +333,53 @@ export class DataTable {
    * @param newColumns recent datatable columns value
    */
   @Watch('columns')
-  columnsChangeHandler(newColumns: DataTableColumn[]) {
+  async columnsChangeHandler(newColumns: DataTableColumn[]) {
+    let hasStructuralChanges = false;
+    const existingColumnSettings = await this.getTableSettings();
     this.columnOrdering(newColumns);
+    const currentColumnSettings = await this.getTableSettings();
+
+    // Observe for structural changes
+    for (const currentColumn in currentColumnSettings) {
```

Comment:
You can use `Object.keys` instead of using `for in` and `hasOwnProperty` check.

Reply:
Yes, but I am failing to see the advantage of suggested alternate solution. Even if I use Object.keys, I still have to loop through.

Reply:
Yes looping is still required but you can avoid the check for `hasOwnProperty`.

Reply:
I am getting Object.keys instead of hasOwnProperty check anyways. Just seems like another way to go about this with no realtime noticeable performance gains.

Besides, for..in still feels like the appropriate method to use in this case. Kindly reconsider.